### DISCUSSION TITLE
Validate form doesn't work with form step

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#6811] [Help] [answered]
Hi!

I am having a problem using the validation of a form with steps. This could be a bug, so here are the steps I followed:
- I create a form with steps (6 in total)
- Below this form I have two buttons: 'Save' and 'Save as complete'

The 'Save as Complete' button should check that the fields are valid and then save the data.
The 'Validate only current step' box is not checked, however when I click on the button and an error is present in another step than the one displayed, the line is saved without any warning. If I change to display a step with an invalid field, the validation works correctly.

I have also tried using a 'Change Form Step' followed by a 'Validate form' (current step only checked) for each step followed by 'Save row' but this does not work. This would allow the first error encountered to be displayed to the user but maybe this is the expected behaviour.

Thanks for your help!

**Comment by melohagan** [comment, marked as answer]
Hi @diogenesbrussels 

This is expected behaviour. A change was made a while back where validation is always run against the current form step only.

This means if you want every form step to be validated, then you will need a **Validate** action on each step navigation button. 

I have raised an issue here: https://github.com/Budibase/budibase/issues/6820
